### ISSUE TITLE
Error: pg: Scan(nil) when scanning JSONB field to interface{}

### ISSUE THREAD

**Issue by joelclouddistrict** [go-pg/pg#299]
Hello, I am not able to retrieve data for the following model:

```
type ModelEvent struct {
	TableName struct{} `sql:"events,alias:events"`
	Id		     int64
	**Metadata     interface{}**
	Created      time.Time
	Updated      time.Time
	Deleted      time.Time `sql:",null"`
}

```
From this table:

```
CREATE TABLE IF NOT EXISTS events (
    id              SERIAL,
    **metadata        JSONB,**
    created         TIMESTAMP NOT NULL DEFAULT NOW(),
    updated         TIMESTAMP NOT NULL DEFAULT NOW(),
    deleted         TIMESTAMP DEFAULT NULL,

    CONSTRAINT events_pkey PRIMARY KEY (id)
)
```

I am able to insert data correctly but when I try to select a list of events with:

```
var mEvents []ModelEvent
q := p.Pool.Model(&mEvents).
      Order("id ASC")
count, err := q.SelectAndCount()
```

The function SelectAndCount() returns the error: **pg: Scan(nil)**. I have traced the error and it happens because the **fieldByIndex** function in line 61 of fields.go returns nil, and then **scanIfaceValue** in scan_value.go verifies that that value is nil and returns the error.

How should I work with JSONB columns? My intention is to have a field with a variable structure.

Thank you.

**Comment by vmihailenco** [comment]
Hello,

Hopefully fixed in v4.8.7. Please try to upgrade.

**Comment by joelclouddistrict** [comment]
Now works correctly, thank you.
